instance/models/mixins/openedx_monitoring.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2016 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Open edX instance monitoring mixin
+"""
+
+# Imports #####################################################################
+
+from django.conf import settings
+from django.db import models
+
+from instance import newrelic
+
+
+# Classes #####################################################################
+
+class OpenEdXMonitoringMixin:
+    """
+    Mixin that sets up availability monitoring for Open edX instances.
+    """
+    def enable_monitoring(self):
+        """
+        Enable monitoring on this instance.
+        """
+        if not settings.NEWRELIC_ADMIN_USER_API_KEY:
+            return
+
```

Comment:
@omarkhan Nit: can we add some logging - `logging.info("Setting up NewRelic monitors")` or something similar? It's more fun to watch a new instance spinning up in the UI if you know what it's doing :)